File: leetcode/C++/find-duplicate-file-in-system.cpp
class Solution {
public:
    vector<vector<string>> findDuplicate(vector<string>& paths) {
        vector<vector<string>> sol;
        unordered_map<string,vector<string>> um;
        string dir,meta,t;
        int p;
        for(string &path : paths){
            dir=path.substr(0,(p=path.find(" ")));
            t=path.substr(p);
            while((p=t.find(" "))>=0){
                meta=t.substr(1,(p=t.find(")")));
                t=t.substr(p+1);
                um[meta.substr((p=meta.find("(")))].push_back(dir+'/'+meta.substr(0,p));
            }
        }
        for(const pair<string, vector<string>> &p : um)
            if(p.second.size()>1) sol.push_back(p.second);
        return sol;
    }
};

File: leetcode/Python/find-duplicate-file-in-system.py
class Solution:
    def findDuplicate(self, paths: List[str]) -> List[List[str]]:
        t, d, sol, m = 0, "", [], collections.defaultdict(list)
        for path in paths:
            for i, s in enumerate(path.split()):
                if i==0:
                    d=s
                else:
                    t=s.find("(")
                    m[s[t:]].append(d+'/'+s[:t])
        for _, v in m.items():
            if len(v) > 1:
                sol.append(v)
        return sol